src/ORM/DataQuery.php

```diff
@@ -655,6 +655,20 @@ public function having($having)
         return $this;
     }
 
+    /**
+     * Add a query to UNION with.
+     *
+     * @param string|null $type One of the SQLSelect::UNION_ALL or SQLSelect::UNION_DISTINCT constants - or null for a default union
+     */
+    public function union(self|SQLSelect $query, ?string $type = null): static
```

Comment:
`self|SQLSelect` why the mixed type? Would prefer to just be a single type if possible

Reply:
This is a `DataQuery` so it makes sense to allow a `DataQuery` as the unioned query. It also makes sense to not _require_ the unioned query to be a `DataQuery`, as a custom `SQLSelect` query is often easier to create.

Reply:
maybe just make it `DataQuery|SQLSelect` so it's a little more readable - we've used `DataQuery` for `subtract()` in this file

Reply:
Done